### PR TITLE
[extension/healthcheckextension] Add response_body configuration option

### LINKED ISSUE #18824: Health Check: Support static response
### Component(s)

extension/healthcheck

### Is your feature request related to a problem? Please describe.

Some load balancer or proxy ([like Envoy](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/health_check.proto#envoy-v3-api-msg-config-core-v3-healthcheck-httphealthcheck)) can be configured to check the response body for a specific match.
There might be cases where a standard response is expected and the load balancer can't easily be changed.
Currently the `healtchcheckextension` returns an empty body which wouldn't work.

### Describe the solution you'd like

Add a new `response_body` field in the configuration of the `healtchcheckextension` to always return that value when it's healthy.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ledor473**
**Description:**
Add `esponse_body` configuration option that allows specifying a specific response body

**Link to tracking Issue:** 
Closes #18824 

**Testing:**
Added tests for both handlers (with / without check collector pipeline)

**Documentation:**
Updated README.md and added an entry in the changelog

### CODE REVIEW

File: extension/healthcheckextension/config.go [hunk 1]
```diff
@@ -32,6 +32,11 @@ type Config struct {
 	// The default path is "/".
 	Path string `mapstructure:"path"`
 
+	// StaticResponseBody represents the body of the response returned by the health check service.
+	// This is override the default response that it would return.
```
Comment (atoulme) [inline extension/healthcheckextension/config.go:36]:
```suggestion
	// This overrides the default response that it would return.
```

File: extension/healthcheckextension/healthcheckextension.go [hunk 2]
```diff
@@ -109,13 +109,34 @@ func (hc *healthCheckExtension) Start(_ context.Context, host component.Host) er
...
+func (hc *healthCheckExtension) baseHandler() http.Handler {
+	if hc.config.StaticResponseBody != "" {
+		return http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+			if hc.state.Get() == healthcheck.Ready {
+				w.WriteHeader(http.StatusOK)
+				if hc.config.StaticResponseBody != "" {
+					w.Write([]byte(hc.config.StaticResponseBody))
+				}
+			} else {
+				w.WriteHeader(http.StatusServiceUnavailable)
```
Comment (atoulme) [inline extension/healthcheckextension/healthcheckextension.go:122]: would it make sense to have a static response body on error too?

File: extension/healthcheckextension/healthcheckextension.go [hunk 3]
```diff
@@ -109,13 +109,34 @@ func (hc *healthCheckExtension) Start(_ context.Context, host component.Host) er
...
 	return http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
 		if hc.check() && hc.state.Get() == healthcheck.Ready {
-			w.WriteHeader(200)
+			w.WriteHeader(http.StatusOK)
+			if hc.config.StaticResponseBody != "" {
+				w.Write([]byte(hc.config.StaticResponseBody))
+			}
 		} else {
-			w.WriteHeader(500)
+			w.WriteHeader(http.StatusServiceUnavailable)
```
Comment (atoulme) [inline extension/healthcheckextension/healthcheckextension.go:139]: That's a 503 vs a 500 - I think that needs to be a separate PR.

File: extension/healthcheckextension/config.go [hunk 4]
```diff
@@ -32,6 +32,11 @@ type Config struct {
 	// The default path is "/".
 	Path string `mapstructure:"path"`
 
+	// StaticResponseBody represents the body of the response returned by the health check service.
```
Comment (atoulme) [inline extension/healthcheckextension/config.go:35]:
```suggestion
	// StaticResponseBody represents the body of the response returned by the health check service when the collector is healthy.
```

File: extension/healthcheckextension/healthcheckextension.go (hunk 2 above)
Comment (ledor473) [reply extension/healthcheckextension/healthcheckextension.go:122]: I wondered the same.

In my case it wasn’t needed because the logic just check if the status is 200 and if the body contains a specific string.
If either are false, it fails the check.

If preferred, I can modify it to something a little more refined like:
```
static_response_code:
  enabled: true/false
  ready: ""
  not_ready: ""
```

Comment (atoulme) [reply extension/healthcheckextension/healthcheckextension.go:122]: I think you can get away with:
```
response_body:
  healthy: ""
  unhealthy: ""
```

No need for enabled, and I renamed the fields:
* static_response_code -> response_body as it is static, anyway, and it's the response body contents
* ready -> healthy since I guess the status can flip back and forth, not just on readiness.

Comment (ledor473) [reply extension/healthcheckextension/healthcheckextension.go:122]: The enabled flag was just to clarify that if either are provided, it would override both responses. 
I’ll update the PR with your suggestion

File: extension/healthcheckextension/healthcheckextension.go (hunk 3 above)
Comment (ledor473) [reply extension/healthcheckextension/healthcheckextension.go]: Good call, reverted

**Comment by ledor473** [comment]
While adding new tests for the `healthy` / `unhealthy`  change, I figured that instead of copy-pasting more repetitive code, I could convert the tests to table-driven tests

**Review by atoulme — APPROVED** [review]
LGTM

**Review by kovrus — APPROVED** [review]
